src/actions/auth.js

```diff
@@ -1,107 +1,142 @@
 import { myFirebase, googleProvider } from '../firebase/firebase';
+import api from '../utils/api';
 
-export const LOGIN_REQUEST = 'LOGIN_REQUEST';
-export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
-export const LOGIN_FAILURE = 'LOGIN_FAILURE';
-export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';
+export const types = {
+  LOGIN_REQUEST: 'LOGIN_REQUEST',
+  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
+  LOGIN_FAILURE: 'LOGIN_FAILURE',
+  SIGNUP_FAILURE: 'SIGNUP_FAILURE',
 
-export const LOGOUT_REQUEST = 'LOGOUT_REQUEST';
-export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
-export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';
+  USER_LOADED: 'USER_LOADED',
+  USER_LOAD_FAILURE: 'USER_LOAD_FAILURE',
 
-export const VERIFY_REQUEST = 'VERIFY_REQUEST';
-export const VERIFY_SUCCESS = 'VERIFY_SUCCESS';
+  LOGOUT_REQUEST: 'LOGOUT_REQUEST',
+  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
+  LOGOUT_FAILURE: 'LOGOUT_FAILURE',
+
+  VERIFY_REQUEST: 'VERIFY_REQUEST',
+  VERIFY_SUCCESS: 'VERIFY_SUCCESS'
+};
 
 const requestLogin = () => {
   return {
-    type: LOGIN_REQUEST
+    type: types.LOGIN_REQUEST
   };
 };
 
 const receiveLogin = user => {
   return {
-    type: LOGIN_SUCCESS,
+    type: types.LOGIN_SUCCESS,
+    user
+  };
+};
+
+const userLoaded = user => {
+  return {
+    type: types.USER_LOADED,
     user
   };
 };
 
+const userLoadError = () => {
+  return {
+    type: types.USER_LOAD_FAILURE
+  };
+};
+
 const loginError = () => {
   return {
-    type: LOGIN_FAILURE
+    type: types.LOGIN_FAILURE
   };
 };
 
 const signUpError = message => {
   return {
-    type: SIGNUP_FAILURE,
+    type: types.SIGNUP_FAILURE,
     message
   };
 };
 
 const requestLogout = () => {
   return {
-    type: LOGOUT_REQUEST
+    type: types.LOGOUT_REQUEST
   };
 };
 
 const receiveLogout = () => {
   return {
-    type: LOGOUT_SUCCESS
+    type: types.LOGOUT_SUCCESS
   };
 };
 
 const logoutError = () => {
   return {
-    type: LOGOUT_FAILURE
+    type: types.LOGOUT_FAILURE
   };
 };
 
 const verifyRequest = () => {
   return {
-    type: VERIFY_REQUEST
+    type: types.VERIFY_REQUEST
   };
 };
 
 const verifySuccess = () => {
   return {
-    type: VERIFY_SUCCESS
+    type: types.VERIFY_SUCCESS
   };
 };
 
+// Load User
+export const loadUser = () => async dispatch => {
+  try {
+    let res = await api.get('/users');
+    if (!res.data) {
+      res = await api.post('/users');
```

Comment:
Why is is there a HTTP post? we're not creating/updating content here

Reply:
we are creating a user in mongo if they don't exist ( this is for extended user details ) and loading them into redux store, this will grab their postcode etc, but just a placeholder for now